shapes.h

```diff
@@ -0,0 +1,57 @@
+/* ----------------------------------------------------------------------------
+ * Copyright &copy; 2015 James Mabe <jamespmabe@gmail.com>
+ * Released under the [MIT License] (http://opensource.org/licenses/MIT)
+ * References: Ben Blazak.
+ * ------------------------------------------------------------------------- */
+
+/**
+ * This header file contains the classes of the shapes
+ */
+
+#ifndef SHAPES_H
+#define SHAPES_H
+
+#include "point.h"
+
+class Shape{
+public:
+    virtual bool contains(const Point & p) const = 0;
+    virtual ~Shape() {}
+    
+};
+
+class Rectangle : public Shape {
+private:
+    Point center;
+    float Rx;
+    float Ry;
+    
+public:
+    Rectangle(Point center = Point(), float Rx = 0, float Ry = 0);
+    
+    bool contains(const Point & p) const;
```

Comment:
no need to redeclare the `contains` method in all subclasses of `Shape`: it's inherited; it just needs to be implemented